name_tracker.go

```diff
@@ -6,14 +6,47 @@ package quamina
 // Quamina's MatchesForEvent API is strongly related to the number of fields it has to look at.  Therefore, a
 // Flattener should call NameTracker for each field it encounters and if it returns false, skip that field
 // and not add it to the flattened output. Here's a clarifying example; consider this JSON event:
-//  {"a": {"b": 1, "c": 2}}
+//
+//	{"a": {"b": 1, "c": 2}}
+//
 // Assuming NameTracker always returns positive, the flattened output should look like
-//  "a\nb" 1
-//  "a\nc" 2
+//
+//	"a\nb" 1
+//	"a\nc" 2
+//
 // Note that the values stored in NameTracker are not the full paths "a\nb" and "a\nc", but the path segments
 // "a", "b", and "c".  This reduces the work the Flattener has to do - whenever it processes an object, it need
 // only look up the member names in the NameTracker. A little thought reveals that this can produce false
 // positives and some potential wasted work, but is a good trade-off.
```

Comment:
Maybe a remark here about the new hierarchical tracker so people understand that NameTracker is not the only optimization in play?

Reply:
@timbray `do we want to keep use of NameTracker and namesUsed ?` what's your opinion on that?